cogs/tags.py

```diff
@@ -157,6 +160,41 @@ def get_tag(self, server, name, *, redirect=True):
                 raise RuntimeError('Tag not found.')
             raise RuntimeError('Tag not found. Did you mean...\n' + '\n'.join(possible_matches))
 
+    async def user_exceeds_tag_limit(self, ctx):
+        """Check to see if user has too many tags.
+
+        Returns:
+        --------
+        bool
+            True if too many tags, False if okay.
+        """
+        owner = ctx.message.author
+        server = ctx.message.server
+
+        if owner.id == ctx.bot.settings.owner:
+            # No limit for bot owner
+            return False
+
+        if server:
+            # No limit for mods and admins of server.
+            mod_role = ctx.bot.settings.get_server_mod(server).lower()
+            admin_role = ctx.bot.settings.get_server_admin(server).lower()
+            roles = [role.name for role in owner.roles]
+            if admin_role in roles or mod_role in roles:
+                return False
+
+        tags = [tag.name for tag in self.config.get('generic', {}).values()
+                if tag.owner_id == owner.id]
+        if server:
+            tags.extend(tag.name for tag in self.config.get(server.id, {}).values()
+                        if tag.owner_id == owner.id)
+        limit = self.settings.get(KEY_MAX, DEFAULT_MAX)
+        if len(tags) >= limit:
+            await self.bot.say("You have too many commands. The maximum number of commands "
+                               "per user is {}, please delete some first!".format(limit))
```

Comment:
Slight nitpick. This is probably more of a helper function right? It says it only returns true/false depending on if they got too many tags, but here it also makes the bot say stuff. Usually, you want to have functions that have a single good purpose, so combining interface output with the boolean output seems odd. It's hard to explain, but this kind of function probably shouldn't worry about what to output to the user.

Reply:
Fair point, I've moved the bot message outside of this helper method.